vortex-array/src/arrays/constant/compute/sum.rs

```diff
@@ -138,21 +137,19 @@ fn sum_float(
     array_len: usize,
     accumulator: &Scalar,
 ) -> VortexResult<Option<f64>> {
-    let v = primitive_scalar
-        .as_::<f64>()
-        .vortex_expect("cannot be null");
-    let array_len = array_len
-        .to_f64()
-        .ok_or_else(|| vortex_err!("array_len must fit the sum type"))?;
-
-    let Ok(array_sum) = v.mul_checked(array_len) else {
-        return Ok(None);
-    };
     let initial = accumulator
         .as_primitive()
         .as_::<f64>()
         .vortex_expect("cannot be null");
-    Ok(Some(initial + array_sum))
+    let v = primitive_scalar
+        .as_::<f64>()
+        .vortex_expect("cannot be null");
+
+    let mut sum = initial;
```

Comment:
can we add a comment for why we do this? i can imagine someone coming along in the future and changing this back to the multiply